rclcpp/test/rclcpp/test_graph_listener.cpp

```diff
@@ -40,15 +40,13 @@ class TestGraphListener : public ::testing::Test
     rclcpp::init(0, nullptr);
     node_ = std::make_shared<rclcpp::Node>(node_name, node_namespace);
 
-    // This dynamic cast is not necessary for the unittests, but instead is used to ensure
-    // the proper type is being tested and covered.
-    node_graph_ =
-      dynamic_cast<rclcpp::node_interfaces::NodeGraph *>(node_->get_node_graph_interface().get());
+    node_graph_ = std::make_shared<rclcpp::node_interfaces::NodeGraph>(
```

Comment:
I think this can simply just be:
```
node_graph_ = node_->get_node_graph_interface();
```

Reply:
I can't assign a `node_graph_interface` to a `node_graph`, am I missing something in your suggestion?

Reply:
I was also thinking you can change the type of node_graph_ to `rclcpp::node_interfaces::NodeGraphInterface`. It's the base type, we don't need to dynamic cast to the derived type for this test.

Reply:
So in this class, the last changes I am looking for would be to adjust the definition of `node_graph_`

```
rclcpp::node_interfaces::NodeGraphInterface::SharedPtr node_graph_
```
Then this line can just be:
```
 node_graph_ = node_->get_node_base_interface();
```

Reply:
Then how I make the method `node_graph()` returning `rclcpp::node_interfaces::NodeGraph *` ?

Reply:
Ok, see 6114d38